src/dominators.rs

```diff
@@ -0,0 +1,277 @@
+//! # The Dominance Relation
+//!
+//! In a directed graph with a root node `R`, a node `A` is said to *dominate* a
+//! node `B` iff every path from `R` to `B` contains `A`.
+//!
+//! The node `A` is daid to *strictly dominate the node `B` iff `A` dominates
+//! `B` and `A != B`.
+//!
+//! The node `A` is said to be the *immediate dominator* of a node `B` iff it
+//! strictly dominates `B` and does not dominate any other nodes which also
+//! dominate `B` in turn.
+
+use std::collections::{HashMap, HashSet};
+use std::hash::Hash;
+
+use visit::{DfsPostOrder, GraphBase, IntoNeighbors, Visitable, Walker};
+
+/// The dominance relation for some graph and root.
+#[derive(Debug, Clone)]
+pub struct Dominators<G>
```

Comment:
Rust has that nice thing that `Dominators` would hold the graph borrowed, so that they are guaranteed to be valid together (because the graph can't change), but at the end of the day, the redundant borrowing decreases the flexibility of the program.